src/analytics_desktop.cc

```diff
@@ -0,0 +1,365 @@
+  }
+  Parameter param(parameter_name, static_cast<int64_t>(parameter_value));
+  LogEvent(name, &param, 1);
+}
+
+void InitiateOnDeviceConversionMeasurementWithEmailAddress(
+    const char* email_address) {
+  (void)email_address;
+  LogWarning("Analytics: InitiateOnDeviceConversionMeasurementWithEmailAddress() is not supported and has no effect on Windows.");
+}
+
+void InitiateOnDeviceConversionMeasurementWithPhoneNumber(
+    const char* phone_number) {
+  (void)phone_number;
+  LogWarning("Analytics: InitiateOnDeviceConversionMeasurementWithPhoneNumber() is not supported and has no effect on Windows.");
+}
+
+void InitiateOnDeviceConversionMeasurementWithHashedEmailAddress(
+    std::vector<unsigned char> hashed_email_address) {
+  (void)hashed_email_address;
+  LogWarning("Analytics: InitiateOnDeviceConversionMeasurementWithHashedEmailAddress() is not supported and has no effect on Windows.");
+}
+
+void InitiateOnDeviceConversionMeasurementWithHashedPhoneNumber(
+    std::vector<unsigned char> hashed_phone_number) {
+  (void)hashed_phone_number;
+  LogWarning("Analytics: InitiateOnDeviceConversionMeasurementWithHashedPhoneNumber() is not supported and has no effect on Windows.");
+}
+
+void SetSessionTimeoutDuration(int64_t milliseconds) {
+  (void)milliseconds;
+  LogWarning("Analytics: SetSessionTimeoutDuration() is not supported and has no effect on Windows.");
+}
+
+Future<std::string> GetAnalyticsInstanceId() {
+  // Not supported by the Windows C API.
+  // Return a Future that is already completed with an error.
+  firebase::FutureHandle handle; // Dummy handle for error
+  // TODO(jules): Ensure g_future_api_table is appropriate or replace with direct Future creation.
+  auto future = MakeFuture<std::string>(&firebase::g_future_api_table, handle);
+  future.Complete(handle, kAnalyticsErrorNotSupportedOnPlatform, "GetAnalyticsInstanceId is not supported on Windows.");
+  return future;
+}
+
+Future<std::string> GetAnalyticsInstanceIdLastResult() {
+  // This typically returns the last result of the async call.
+  // Since GetAnalyticsInstanceId is not supported, this also returns a failed future.
+  firebase::FutureHandle handle;
+  auto future = MakeFuture<std::string>(&firebase::g_future_api_table, handle);
+  future.Complete(handle, kAnalyticsErrorNotSupportedOnPlatform, "GetAnalyticsInstanceId is not supported on Windows.");
+  return future;
+}
+
+Future<int64_t> GetSessionId() {
+  // Not supported by the Windows C API.
+  firebase::FutureHandle handle;
+  auto future = MakeFuture<int64_t>(&firebase::g_future_api_table, handle);
+  future.Complete(handle, kAnalyticsErrorNotSupportedOnPlatform, "GetSessionId is not supported on Windows.");
+  return future;
+}
+
+Future<int64_t> GetSessionIdLastResult() {
+  firebase::FutureHandle handle;
+  auto future = MakeFuture<int64_t>(&firebase::g_future_api_table, handle);
+  future.Complete(handle, kAnalyticsErrorNotSupportedOnPlatform, "GetSessionId is not supported on Windows.");
```

Comment:
For all of these methods with warnings or logs about not being supported on Windows, could you change it to "on Desktop" instead?